xstate-chatbot/nodejs/src/machine/vitals.js

```diff
@@ -55,6 +98,106 @@ const vitalsFlow = {
+      initial: 'prompt',
+      states: {
+        prompt: {
+          onEntry: assign((context, event) => {
+            dialog.sendMessage(context, dialog.get_message(messages.rrtMobileNumber.prompt, context.user.locale));
+          }),
+          on: {
+            USER_MESSAGE: 'process'
+          }
+        },
+        process: {
+          onEntry: assign((context, event) => {
+            context.message = dialog.get_input(event, false);
+          }),
+          invoke: {
+            src: (context, event) => vitalsService.getPatientDetailsFromMobileNumber(context.message),
+            onDone: [
+              {
+                cond: (context, event) => event.data.data === null,
+                actions: assign((context, event) => {
+                  context.persons = event.data;
+                }),
+                target: 'error'
+              },
+              {
+                cond: (context, event) => event.data.data.length >= 1,
+                actions: assign((context, event) => {
+                  console.log(event.data.data.length)
+                  context.persons = event.data;
+                }),
+                target: '#selectPerson'
+              },
+              {
+                target: 'error'
+              }
+            ]
+          }
+        },
+        error: {
+          onEntry: assign((context, event) => {
+            dialog.sendMessage(context, dialog.get_message(messages.noUserFound, context.user.locale), false);
+          }),
+          always: 'prompt'
+        }
+      }
+    },
+    selectPerson: {
+      id: 'selectPerson',
+      initial: 'prompt',
+      states: {
+        prompt: {
+          onEntry: assign((context, event) => {
+            let message = dialog.get_message(messages.selectPerson, context.user.locale);
+            let grammer = [];
+
+            for (let i = 0; i < event.data.data.length; i++) {
+              let person = event.data.data[i];
+
+              let grammerItem = { intention: event.data.data[i], recognize: [(i + 1).toString()] };
+
+              grammer.push(grammerItem);
+              message += '\n' + (i + 1) + '. ' + event.data.data[i].MASTER_NAME + '\n  com_status : '+event.data.data[i].com_status+'\n  fateh_kit_delivered : '+event.data.data[i].fateh_kit_delivered;
+            }
+            context.grammer = grammer;
+            dialog.sendMessage(context, message);
+          }),
+          on: {
+            USER_MESSAGE: 'process'
+          }
+        },
+        process: {
+          onEntry: assign((context, event) => {
+            context.intention = dialog.get_intention(context.grammer, event, true);
+          }),
+          always: [
+            {
+              cond: (context) => context.intention == dialog.INTENTION_UNKOWN,
+              target: 'error'
+            },
+            {
+              actions: assign((context, event) => {
+                context.slots.vitals.MASTER_ID = context.intention.MASTER_ID;
```

Comment:
Store these 3 attributes in: context.slots.person.MAS...
So someone doesn't confuse the fateh_kit_delivered(data from cova) with the question that is being asked below(FatehKitDelivered)

Reply:
Please make the suggested change

Reply:
made the changes as you suggested